packages/near-membrane-base/src/__tests__/environment.spec.ts

```diff
@@ -152,14 +153,279 @@ describe('VirtualEnvironment', () => {
 
             ve.remap(globalThis, getFilteredEndowmentDescriptors(endowments));
 
-            expect(globalThis.p).toBe(1); // count + 1
-            expect(globalThis.p).toBe(1); // count + 1
-            expect(globalThis.p).toBe(1); // count + 1
+            expect(globalThis.c).toBe(1); // count + 1
+            expect(globalThis.c).toBe(1); // count + 1
+            expect(globalThis.c).toBe(1); // count + 1
             expect(count).toBe(3);
-            globalThis.p = 99;
-            expect(globalThis.p).toBe(1);
+            globalThis.c = 99;
+            expect(globalThis.c).toBe(1);
             expect(blueSetValue).toBe(99);
             expect(count).toBe(6);
         });
     });
+    describe('VirtualEnvironment.prototype.lazyRemap', () => {
+        it('Skips index-like properties', () => {
+            // Ignoring "Property 'assertions' does not exist on type '{...}'."
+            // @ts-ignore
+            expect.assertions(1);
+
+            // eslint-disable-next-line no-eval
+            const redConnector = globalThis.eval(initSourceTextInStrictMode);
+
+            const ve = new VirtualEnvironment({
+                blueConnector: init,
+                redConnector,
+                distortionCallback(v) {
+                    return v;
+                },
+            });
+
+            const redValue = {};
+            const endowments = {
+                0: 'foo',
+            };
+
+            ve.lazyRemap(redValue, Object.keys(getFilteredEndowmentDescriptors(endowments)));
+            expect(Object.getOwnPropertyNames(redValue).length).toBe(0);
+        });
+
```

Comment:
☝️test `'Skips index-like properties'` looks to be covered by `'Skips index-like properties, called after remap'` 👇